src/tile.cpp

```diff
@@ -1048,9 +1043,9 @@ void Tile::replaceThing(uint32_t index, Thing* thing)
 	}
 }
 
-void Tile::removeThing(Thing* thing, uint32_t count)
+void Tile::removeThing(const std::shared_ptr<Thing>& thing, uint32_t count)
 {
-	Creature* creature = thing->getCreature();
+	const auto& creature = thing->getCreature();
```

Comment:
`Creature` and `Item`.

```suggestion
	if (const auto& creature = thing->getCreature()) {
		CreatureVector* creatures = getCreatures();
		if (creatures) {
			auto it = std::find(creatures->begin(), creatures->end(), thing);
			if (it != creatures->end()) {
				g_game.map.clearSpectatorCache();
				if (creature->getPlayer()) {
					g_game.map.clearPlayersSpectatorCache();
				}

				creatures->erase(it);
			}
		}
	} else if (const auto& item = thing->getItem()) {
```